insolar/record/record.pb.go

```diff
@@ -0,0 +1,5180 @@
+
+type ObjectDeactivateRecord struct {
+	Domain    github_com_insolar_insolar_insolar.Reference `protobuf:"bytes,1,opt,name=Domain,proto3,customtype=github.com/insolar/insolar/insolar.Reference" json:"Domain"`
+	Request   github_com_insolar_insolar_insolar.Reference `protobuf:"bytes,2,opt,name=Request,proto3,customtype=github.com/insolar/insolar/insolar.Reference" json:"Request"`
+	PrevState github_com_insolar_insolar_insolar.ID        `protobuf:"bytes,3,opt,name=PrevState,proto3,customtype=github.com/insolar/insolar/insolar.ID" json:"PrevState"`
+}
+
+func (m *ObjectDeactivateRecord) Reset()      { *m = ObjectDeactivateRecord{} }
+func (*ObjectDeactivateRecord) ProtoMessage() {}
+func (*ObjectDeactivateRecord) Descriptor() ([]byte, []int) {
+	return fileDescriptor_0c86cc3f6f53fe45, []int{9}
+}
+func (m *ObjectDeactivateRecord) XXX_Unmarshal(b []byte) error {
+	return m.Unmarshal(b)
+}
+func (m *ObjectDeactivateRecord) XXX_Marshal(b []byte, deterministic bool) ([]byte, error) {
+	if deterministic {
+		return xxx_messageInfo_ObjectDeactivateRecord.Marshal(b, m, deterministic)
+	} else {
+		b = b[:cap(b)]
+		n, err := m.MarshalTo(b)
+		if err != nil {
+			return nil, err
+		}
+		return b[:n], nil
+	}
+}
+func (m *ObjectDeactivateRecord) XXX_Merge(src proto.Message) {
+	xxx_messageInfo_ObjectDeactivateRecord.Merge(m, src)
+}
+func (m *ObjectDeactivateRecord) XXX_Size() int {
+	return m.Size()
+}
+func (m *ObjectDeactivateRecord) XXX_DiscardUnknown() {
+	xxx_messageInfo_ObjectDeactivateRecord.DiscardUnknown(m)
+}
+
+var xxx_messageInfo_ObjectDeactivateRecord proto.InternalMessageInfo
+
+type Record struct {
+	// Types that are valid to be assigned to Union:
+	//	*Record_Genesis
+	//	*Record_Child
+	//	*Record_Jet
+	//	*Record_Request
+	//	*Record_Result
+	//	*Record_Type
+	//	*Record_Code
+	//	*Record_ObjectActivate
+	//	*Record_ObjectAmend
+	//	*Record_ObjectDeactivate
+	Union isRecord_Union `protobuf_oneof:"union"`
+}
+
+func (m *Record) Reset()      { *m = Record{} }
+func (*Record) ProtoMessage() {}
+func (*Record) Descriptor() ([]byte, []int) {
+	return fileDescriptor_0c86cc3f6f53fe45, []int{10}
+}
+func (m *Record) XXX_Unmarshal(b []byte) error {
+	return m.Unmarshal(b)
+}
+func (m *Record) XXX_Marshal(b []byte, deterministic bool) ([]byte, error) {
+	if deterministic {
+		return xxx_messageInfo_Record.Marshal(b, m, deterministic)
+	} else {
+		b = b[:cap(b)]
+		n, err := m.MarshalTo(b)
+		if err != nil {
+			return nil, err
+		}
+		return b[:n], nil
+	}
+}
+func (m *Record) XXX_Merge(src proto.Message) {
+	xxx_messageInfo_Record.Merge(m, src)
+}
+func (m *Record) XXX_Size() int {
+	return m.Size()
+}
+func (m *Record) XXX_DiscardUnknown() {
+	xxx_messageInfo_Record.DiscardUnknown(m)
+}
+
+var xxx_messageInfo_Record proto.InternalMessageInfo
+
+type isRecord_Union interface {
+	isRecord_Union()
+	Equal(interface{}) bool
+	MarshalTo([]byte) (int, error)
+	Size() int
+}
+
+type Record_Genesis struct {
```

Comment:
Can we control these names?

Reply:
We can, but we don't need to. It won't be used outside "compatibility layer".